sdk/identity/azure-identity/tests/test_auth_file_credential.py

```diff
@@ -0,0 +1,67 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+from azure.identity import AuthFileCredential
+from azure.core.exceptions import ClientAuthenticationError
+
+from helpers import validating_transport, Request, mock_response, build_aad_response
+
+import pytest
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+
+
+def test_auth_file_credential_parse():
+    client_id = 'mockclientid'   # same value with that in authfile.json
+    secret = 'mockclientsecret'  # same value with that in authfile.json
+    tenant_id =  'mocktenantid'  # same value with that in authfile.json
+    access_token = '***'
+
+    transport = validating_transport(
+        requests=[Request(url_substring=tenant_id, required_data={"client_id": client_id, "client_secret": secret})],
+        responses=[
+            mock_response(
+                json_payload={
+                    "token_type": "Bearer",
+                    "expires_in": 42,
+                    "ext_expires_in": 42,
+                    "access_token": access_token
+                }
+            )
+        ]
+    )
+
+    token = AuthFileCredential('{}/authfile.json'.format(os.path.dirname(__file__)), transport=transport).get_token('scope')
```

Comment:
[mock_open](https://docs.python.org/3/library/unittest.mock.html#mock-open) would be tidier and more flexible than reading actual files.